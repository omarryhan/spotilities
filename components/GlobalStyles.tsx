import { createGlobalStyle } from 'styled-components';

export default createGlobalStyle`
    @font-face {
        font-family: 'Proxima Nova';
        src: url('/fonts/prox.eot');
        src: url('/fonts/prox.woff') format('woff'),
            url('/fonts/prox.ttf') format('truetype');
        font-weight: normal;
        font-style: normal;
    }

    @font-face {
        font-family: 'Proxima Thin';
        src: url('/fonts/proxthin.eot');
        src: url('/fonts/proxthin.woff') format('woff'),
            url('/fonts/proxthin.ttf') format('truetype');
        font-weight: normal;
        font-style: normal;
    }

    @font-face {
        font-family: 'Proxima Bold';
        src: url('/fonts/proxbold.eot');
        src: url('/fonts/proxbold.woff') format('woff'),
            url('/fonts/proxbold.ttf') format('truetype');
        font-weight: normal;
        font-style: normal;
    }

    * {
        box-sizing: border-box;
        font-display: swap;
        color: white;
        font-family: Proxima Nova, Georgia, sans-serif;
    }

    html {
        /* font-size: calc(16px + 6 * ((100vw - 320px) / 680)); */
        /* calc([minimum size] + ([maximum size] - [minimum size]) * ((100vw - [minimum viewport width]) / ([maximum viewport width] - [minimum viewport width]))); */
        font-size: calc(16px + (24 - 16) * (100vw - 300px) / (1600 - 300));
    }

    body {
        -webkit-font-smoothing: antialiased;
        -moz-osx-font-smoothing: grayscale;
        background-color: rgba(18, 18, 18, 1);
    }

    html, body {
        margin: 0;
        padding: 0;
    }
`;
