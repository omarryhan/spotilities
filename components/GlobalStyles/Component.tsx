import { createGlobalStyle } from 'styled-components';

export default createGlobalStyle`
    @font-face {
        font-family: 'Proxima Nova';
        src: url('/fonts/prox.eot');
        src: url('/fonts/prox.woff') format('woff'),
            url('/fonts/prox.ttf') format('truetype');
        font-weight: normal;
        font-style: normal;
    }

    @font-face {
        font-family: 'Proxima Thin';
        src: url('/fonts/proxthin.eot');
        src: url('/fonts/proxthin.woff') format('woff'),
            url('/fonts/proxthin.ttf') format('truetype');
        font-weight: normal;
        font-style: normal;
    }

    @font-face {
        font-family: 'Proxima Bold';
        src: url('/fonts/proxbold.eot');
        src: url('/fonts/proxbold.woff') format('woff'),
            url('/fonts/proxbold.ttf') format('truetype');
        font-weight: normal;
        font-style: normal;
    }

    * {
        box-sizing: border-box;
        font-display: swap;
        color: white;
        font-family: Proxima Nova, Georgia, sans-serif;
    }

    html, body {
        margin: 0;
        padding: 0;
    }

    html {
        font-size: calc(16px + (24 - 16) * (100vw - 300px) / (1600 - 300));
    }

    body {
        -webkit-font-smoothing: antialiased;
        -moz-osx-font-smoothing: grayscale;
        margin: 0;
        padding: 0;
    }
`;
