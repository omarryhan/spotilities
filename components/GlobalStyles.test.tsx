import React from 'react';
import renderer from 'react-test-renderer';
import GlobalStyles from './GlobalStyles';
import 'jest-styled-components';

describe('GlobalStyles', () => {
  it('is unchanged', () => {
    const tree = renderer.create(<GlobalStyles />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
