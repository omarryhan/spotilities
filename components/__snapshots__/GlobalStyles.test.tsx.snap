// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalStyles is unchanged 1`] = `null`;
